integr8ly/integreatly-operator
# Mgdapi 3695: Add mertics for SRE Dashboard

# Issue link

https://issues.redhat.com/browse/MGDAPI-3695

# What

Adding new alerts for use in SRE dashboards.
* cluster installation type
* addon version and status
* alert summary which excludes the always firing DMS alert

# Verification steps



* prepare to install operator via bundle (metrics are create in the operator pod)
* in openshift UI under Observe/mertrics set up refreshing metrics for `rhoam_version`, `rhoam_alerts` and `rhoam_cluster`
* install the operator
* Expected:
  * `rhoam_cluster` shows the following
    * type = OSD or ROSA
    * externalID = string that match ID in ocm
    * version = matches openshift version of cluster
  * `rhoam_version` has a status field of "Installing"
  * `rhoam_alerts` Once the Observability Operator has being installed and is up in running. The metric should create a summary of alerts pending and firing. Ie. PodCrashLoops should only being seen once even if there are multiple alerts firing. The total count of alerts should be listed.
  
* Once installation is complete
* Expected: `rhoam_version` has a status of "installed"

* Create an upgrade for the bundle and install the upgrade
* Expected: During the upgrade the status in `rhoam_version` changes to "Upgrade"
